### PR TITLE
feat: layout

### PR DESCRIPTION
**Pull request by m0ksem**
NEW PR with Sidebar and Navbar fixes

Introduces breaking changes in Navbar

closes https://github.com/epicmaxco/vuestic-ui/issues/1430
closes https://github.com/epicmaxco/vuestic-ui/issues/3496

Features:
- [x] New component: VaLayout
- [x] top, bottom, left, right slots
- [x] order
- [x] animations support
- [x] absolute areas
- [x] fixes areas
- [x] mobile friendly
- [x] docs and examples
- [x] window scroll (see https://github.com/epicmaxco/vuestic-admin/pull/927#discussion_r1273842969)

Default:
image

Order:
image

Absolute:
image

### CODE REVIEW

File: packages/docs/page-config/ui-elements/layout/index.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+export default definePageConfig({
+  blocks: [
+    block.title('Layout'),
+    block.paragraph('Component is used for building App layout. It is based on CSS Grid and provides a simple API for building complex layouts.'),
+    block.paragraph(`
+VaLayout component is used in pair with [VaSidebar](/ui-elements/sidebar)[[target=_blank]], [VaNavbar](/ui-elements/navbar)[[target=_blank]] components.
```
Comment (asvae) [inline packages/docs/page-config/ui-elements/layout/index.ts:6]:
```suggestion
VaLayout component could be used in pair with [VaSidebar](/ui-elements/sidebar)[[target=_blank]], [VaNavbar](/ui-elements/navbar)[[target=_blank]] components.
```

File: packages/docs/page-config/ui-elements/navbar/examples/Default.vue [hunk 2]
```diff
@@ -10,10 +10,11 @@
         Right slot
       </va-navbar-item>
     </template>
-
-    <va-navbar-item class="navbar-item-slot">
-      Default slot (center)
-    </va-navbar-item>
+    <template #center>
```
Comment (asvae) [inline packages/docs/page-config/ui-elements/navbar/examples/Default.vue:13]: Menu button does nothing:

image

File: packages/docs/page-config/ui-elements/layout/examples/AllSlots.vue [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<script setup lang="ts">
+  const showLeftSidebar = ref(true)
+  const showRightSidebar = ref(true)
+</script>
+
+<template>
```
Comment (asvae) [inline packages/docs/page-config/ui-elements/layout/examples/AllSlots.vue:6]: Would be nice to have something that looks all right on mobile:

image

File: packages/docs/page-config/ui-elements/layout/examples/Order.vue [hunk 4]
```diff
@@ -0,0 +1,102 @@
+<script setup lang="ts">
```
Comment (asvae) [inline packages/docs/page-config/ui-elements/layout/examples/Order.vue:1]: Also would be nice to have a plausible mobile example.

image

Maybe we can just make sidebars proportional on mobile?

**Review by asvae — CHANGES_REQUESTED** [review]
Found a couple of bugs.

I checked experimental.

**Comment by m0ksem** [comment]
Wondering if we want to improve Navbar so it can have shadow and bordered class.

Also, looks like having VaFooter (the same as Sidebar, but with top shadow and border) will be a huge DX improvement.

**Comment by asvae** [comment]
Moved story issue here - #3734

File: packages/docs/page-config/ui-elements/layout/examples/Default.vue [hunk 5]
```diff
@@ -0,0 +1,77 @@
+<script setup lang="ts">
```
Comment (asvae) [inline packages/docs/page-config/ui-elements/layout/examples/Default.vue:1]: Text eloped

image

**Comment by asvae** [comment]
@m0ksem button visual bug is still relevant

I suggest we create a button with right 5rem or so, so that it doesn't overlap with close.
We also probably want to make this button have white half-transparent background and make it stick on scroll.

image

**Review by asvae — APPROVED** [review]

**Pull request merged**